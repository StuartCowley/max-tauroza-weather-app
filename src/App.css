@import url('https://fonts.googleapis.com/css2?family=Goldman&display=swap');

.title {
  font-family: 'Goldman';
}
