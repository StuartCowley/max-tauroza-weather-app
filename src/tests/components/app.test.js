import React from "react";
import { render, screen } from "@testing-library/react";
import axios from "axios";
import App from "../../components/app";

describe("App", () => {
  it("renders app components correctly", () => {
    const { asFragment } = render(<App />);

    expect(asFragment()).toMatchSnapshot();
  });
});

describe("getForecast method", () => {
  it("gets correct data in response", async () => {
    const mockedData = {
      location: {
        city: "Test city",
        country: "Test country",
      },
      forecasts: [
        {
          date: 111111,
          temperature: { max: 2, min: 1 },
          wind: { speed: 1, direction: "test direction" },
          humidity: 1,
          description: "Test description",
          icon: 1,
        },
        // {
        //   date: 222222,
        //   temperature: { max: 3, min: 2 },
        //   wind: { speed: 2, direction: "test direction 2" },
        //   humidity: 2,
        //   description: "Test description 2",
        //   icon: 2,
        // },
      ],
    };

    jest.spyOn(axios, "get").mockResolvedValue(mockedData);
    render(<App />);

    expect(await screen.findByText("test direction")).toBeInTheDocument();
  });
});
