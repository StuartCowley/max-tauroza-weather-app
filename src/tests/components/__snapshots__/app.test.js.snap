// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<DocumentFragment>
  <div
    class="weather-app"
  >
    <h1>
      Manchester, UK
    </h1>
    <div
      class="forecast-summary-list"
    >
      <div
        class="forecast-summary"
        data-testid="forecast-summary"
      >
        <div
          class="forecast-summary_date"
        >
          Thu Jan 01 1970
        </div>
        <div
          class="forecast-summary_icon"
          data-testid="forecast-icon"
        >
          <img
            alt="Weather icon"
            src="https://cdn-icons-png.flaticon.com/128/365/365851.png"
          />
        </div>
        <div
          class="forecast-summary_temperature"
        >
          22째C
        </div>
        <div
          class="forecast-summary_description"
        >
          Stub description
        </div>
        <button
          type="button"
        >
          More details
        </button>
      </div>
      <div
        class="forecast-summary"
        data-testid="forecast-summary"
      >
        <div
          class="forecast-summary_date"
        >
          Thu Jan 01 1970
        </div>
        <div
          class="forecast-summary_icon"
          data-testid="forecast-icon"
        >
          <img
            alt="Weather icon"
            src="https://cdn-icons-png.flaticon.com/128/365/365925.png"
          />
        </div>
        <div
          class="forecast-summary_temperature"
        >
          22째C
        </div>
        <div
          class="forecast-summary_description"
        >
          Stub description
        </div>
        <button
          type="button"
        >
          More details
        </button>
      </div>
    </div>
    <div
      class="forecast-details"
    >
      <h2
        class="forecast-details_date"
      >
        Thu Jan 01 1970
      </h2>
      <div
        class="forecast-details_maximum_temperature"
      >
        Max Temperature: 22째C
      </div>
      <div
        class="forecast-details_minimum_temperature"
      >
        Min Temperature: 12째C
      </div>
      <div
        class="forecast-details_humidity"
      >
        Humidity: 30
      </div>
      <div
        class="forecast-details_wind"
      >
        Wind: 13Stub direction
      </div>
    </div>
  </div>
</DocumentFragment>
`;
